src/EditorFeatures/Core/Implementation/FindReferences/IFindReferencesNavigableItemResultProvider.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.Navigation;
+
+namespace Microsoft.CodeAnalysis.Editor
+{
+    internal interface IFindReferencesResultProvider
```

Comment:
I would say FindReferencesService is analogous to CodeFixService and FindReferencesResultProvider is analogous to CodeFixProvider. The service basically drives all the providers and then is responsible for aggregating/displaying the results. The provider itself should be only data-oriented and eventually we will change these result provider interfaces to abstract types just like CodeFixProvider